docs/guide.md

```diff
@@ -344,6 +344,39 @@ func injectReader() io.Reader {
 }
 ```
 
+### Use Fields of a Struct as Providers
+
+Sometimes the providers the user wants are some fields of a struct. Instead of
+writing "extraction" helper functions to provide each individual field, `wire`
```

Comment:
I think it would be helpful to work out the example of the helper functions, because otherwise this paragraph is a bit abstract.

Reply:
I added a small example of the helper function and also some other fields. I didn't want to make the example too long so I didn't use the other fields.

Reply:
Right. I guess I was thinking a flow more like:

If you find yourself writing a provider like `getS` in the example below to promote struct fields into provided types:

```go
type Foo struct {
  S string
  N int
  F float64
}

func getS(foo Foo) string {
  return foo.S
}
```

You can instead use `wire.FieldsOf` to use those fields directly:

(Example you have now.)

The generated injector would look like this:

(Example you have now.)

You can add as many field names to a `wire.FieldsOf` call as you like.

Reply:
Done.